client/src/components/Booking.js

```diff
@@ -0,0 +1,162 @@
+    [theme.breakpoints.between("xs", "sm")]: {
+      gridRow: "4 / span 1",
+      gridColumn: "1 / span 1",
+    },
+  },
+  button: {
+    marginLeft: 10,
+  },
+}));
+
+const formatDate = dateStr => {
+  const date = new Date(dateStr)
+  return `${doubleNumber(date.getMonth() + 1)}/${doubleNumber(date.getDate())} ${doubleNumber(date.getHours())}:${doubleNumber(date.getMinutes())}`
+}
+
+const doubleNumber = number => (
+  number < 10 ? `0${number}` : number
+)
+
+function Booking({ _id, isBooking, isMyJobs, sitter_id, user_id, start, end, paid }) {
+  const classes = useStyles();
+  const { profile, setProfile } = useProfileContext();
+  const [src, setSrc] = useState(null);
+  const [name, setName] = useState("");
+  useEffect(() => {
+      const id = isMyJobs ? user_id : sitter_id
+      fetch(`/profile/${id}`).then(response => {
+        response.json().then(profile => {
+          setSrc(profile.photo);
+          setName(`${profile.firstName} ${profile.lastName}`);
+        })
+      })
+  },[setSrc, isMyJobs, sitter_id, user_id])
+  const handleAccept = () => {
+    handleAcceptOrDecline(true)
+  }
+  const handleDecline = () => {
+    handleAcceptOrDecline(false)
+  }
+  const handleAcceptOrDecline = accept => {
+    const newProfile = {...profile}
+    let updatedRequest
+    newProfile.requests.forEach(request => {
+      if (request._id === _id) {
+        request.accepted = accept
+        request.declined = !accept
+        updatedRequest = request
+      }
+    })
+    setProfile(newProfile)
+    const options = {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(updatedRequest),
+    };
+    fetch(`/request/update/${updatedRequest._id}`, options);
+  }
+  return (
+    <Card variant="outlined" className={classes.booking}>
+      <Avatar src={src} className={classes.photo} />
+      <h2 className={classes.name}>{name}</h2>
+      <h3 className={classes.date}>
+        {formatDate(start)} - {formatDate(end)}
+      </h3>
+      <div className={classes.buttons}>
+        {!isBooking && !isMyJobs ? (
+          <Button
+            onClick={handleDecline}
+            color="primary"
+            variant="contained"
+            className={classes.button}
+          >
+            Cancel
+          </Button>
+        ) : (
+          <Fragment>
+            <Button
+              onClick={isBooking ? null : handleAccept}
+              color="primary"
+              variant="contained"
+              className={classes.button}
+            >
+              {isBooking ? "Message" : "Accept"}
+            </Button>
+            <Button
+              onClick={
+                isBooking ? (paid ? handleDecline : null) : handleDecline
+              }
+              color="primary"
+              variant="contained"
+              className={classes.button}
+            >
+              {isBooking ? (isMyJobs || paid ? "Cancel" : "Pay") : "Decline"}
```

Comment:
Interesting bug here, if someone accepts a request from themselves, it actually shows two cancel buttons. Not high priority because an edge case but something to note.

Reply:
Do you mean if you make a request to yourself? Yeah I think we should update the request controllers so that isn't even possible in the first place